consensus/safety-rules/src/safety_rules.rs

```diff
@@ -152,23 +161,34 @@ impl SafetyRules {
     /// Check if the executed result extends the parent result.
     fn extension_check(&self, vote_proposal: &VoteProposal) -> Result<VoteData, Error> {
         let proposed_block = vote_proposal.block();
-        let new_tree = vote_proposal
-            .accumulator_extension_proof()
-            .verify(
-                proposed_block
-                    .quorum_cert()
-                    .certified_block()
-                    .executed_state_id(),
-            )
-            .map_err(|e| Error::InvalidAccumulatorExtension(e.to_string()))?;
-        Ok(VoteData::new(
-            proposed_block.gen_block_info(
-                new_tree.root_hash(),
-                new_tree.version(),
-                vote_proposal.next_epoch_state().cloned(),
-            ),
-            proposed_block.quorum_cert().certified_block().clone(),
-        ))
+        if !self.decoupled_execution {
```

Comment:
we should keep the extension_check and bypass it in the caller, and add the extension_check in sign_commit_vote later

Reply:
resolved